golang/connection.go

```diff
@@ -418,22 +418,10 @@ func (c *Connection) withStateRLock(f func() error) error {
 // send the incoming frame to a channel; the init handlers are a notable exception, since we cannot
 // process new frames until the initialization is complete.
 func (c *Connection) readFrames() {
-	fhBuf := typed.NewReadBufferWithSize(FrameHeaderSize)
-
 	for {
-		if _, err := fhBuf.FillFrom(c.conn, FrameHeaderSize); err != nil {
-			c.connectionError(err)
-			return
-		}
-
 		frame := c.framePool.Get()
-		if err := frame.Header.read(fhBuf); err != nil {
-			// TODO(mmihic): Should be a protocol error
-			c.connectionError(err)
-			return
-		}
-
-		if _, err := c.conn.Read(frame.SizedPayload()); err != nil {
+		if err := frame.ReadFrom(c.conn); err != nil {
+			c.framePool.Release(frame)
```

Comment:
Does the frame not need to be released for the success case when the handlers are called?